### PR TITLE
Swiping Index Fix

### PR DESCRIPTION
**Pull request by jjennifergu**
## Overview

Fix "104!West issue" where the same eatery shows up through all buttons from the "Nearest to You" section.

## Changes Made

- Pass in location sorted eatery array to `EateryPageViewController`
- Loop through eateries by index and pass index into `EateryPageViewController` to begin on the right page

## Test Coverage

Playtesting

## Next Steps

Make sure index is correct when swiping is implemented for all eateries.

## Screenshots

<details>

  <summary>Before</summary>

  

https://user-images.githubusercontent.com/57200368/232289133-d8afa5a2-b828-4a3a-b310-c3f7766abfbc.mp4
</details>
<details>
  <summary>After</summary>

  

  

https://user-images.githubusercontent.com/57200368/232289143-64c1c52b-9db8-426f-bba4-008dfd8cebf0.mp4

</details>

### CODE REVIEW

File: EateryModel/Sources/EateryModel/Networking/EateryAPI.swift [hunk 1]
```diff
@@ -22,13 +22,9 @@ public struct EateryAPI {
...
         let (data, _) = try await URLSession.shared.data(from: url, delegate: nil)
-        let schemaApiResponse = try decoder.decode(Schema.APIResponse.self, from: data)
+        let schemaApiResponse = try decoder.decode([Schema.Eatery].self, from: data)
 
-        if let error = schemaApiResponse.error {
-            throw EateryAPIError.apiResponseError(error)
-        }
-
-        return schemaApiResponse.data.map(SchemaToModel.convert)
+        return schemaApiResponse.map(SchemaToModel.convert)
```
Comment (NgaiJustin) [inline EateryModel/Sources/EateryModel/Networking/EateryAPI.swift:27]: Is there a reason why the error handling here was deleted?

**Review by NgaiJustin — CHANGES_REQUESTED** [review]
Most of the changes in this PR seem to be the same as #14, not sure if you want to merge this one or that one first. Also, for the decoding in `EateryAPI.swift`, I think you might want to add some form of error handling for easier debugging in the future.

File: EateryModel/Sources/EateryModel/Networking/Schema.swift [hunk 2]
```diff
@@ -63,19 +51,21 @@ internal enum Schema {
...
+        
+        internal let eatery: Int? // Need to communicate with backend about purpose of this var
 
-        internal let canonicalDate: String
-
-        internal let description: String?
-
-        internal let endTimestamp: Int
+        internal let end: Int?
+
```
Comment (ReadeP) [inline EateryModel/Sources/EateryModel/Networking/Schema.swift:63]: remove tabbing

File: EateryModel/Sources/EateryModel/Networking/Schema.swift [hunk 3]
```diff
@@ -63,19 +51,21 @@ internal enum Schema {
...
 
-        internal let canonicalDate: String
-
-        internal let description: String?
-
-        internal let endTimestamp: Int
+        internal let end: Int?
+        
+        internal let eventDescription: String?
+
```
Comment (ReadeP) [inline EateryModel/Sources/EateryModel/Networking/Schema.swift:65]: removing tabbing

**Review by ReadeP — APPROVED** [review]
Looks good - just address what Justin is talking about

File: EateryModel/Sources/EateryModel/Networking/EateryAPI.swift (hunk 1 above)
Comment (jjennifergu) [reply EateryModel/Sources/EateryModel/Networking/EateryAPI.swift:27]: you're totally right i shouldn't have deleted it. will add it back and request review again

File: Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift [hunk 4]
```diff
@@ -47,13 +47,12 @@ class EateryPageViewController: UIPageViewController {
...
     private func setUpPages() {
-        let sortedEateries = eateries.sorted(by: { $0.index < $1.index })
-        sortedEateries.forEach { eatery in
+       eateries.forEach { eatery in
             let eateryVC = EateryModelController()
             eateryVC.setUp(eatery: eatery)
             pages.append(eateryVC)
         }
-        setViewControllers([pages[selectedIndex]], direction: .forward, animated: true, completion: nil)
+        setViewControllers([pages[index]], direction: .forward, animated: true, completion: nil) // fix index bug
```
Comment (NgaiJustin) [inline Eatery Blue/Eatery Blue/UI/EateryScreen/EateryPageViewController.swift:55]: Is this comment a TODO? If it isn't you might want to consider removing this.

**Review by NgaiJustin — APPROVED** [review]
Yup 🚀

**Pull request merged**